File: app/src/main/java/com/fpondarts/foodie/controller/UserDataActivity.java
package com.fpondarts.foodie.controller;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.media.Image;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

import com.fpondarts.foodie.R;
import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.auth.FirebaseUser;

public class UserDataActivity extends AppCompatActivity {

    private TextView mTvFullname;
    private TextView mTvEmail;
    //private TextView mTvRole;



    private Button mSignOutButton;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_user_data);

        Intent intent = getIntent();
        mTvFullname = (TextView) findViewById(R.id.tvFullName);
        mTvEmail = (TextView) findViewById(R.id.tvEmail);
        //mTvRole = (TextView) findViewById(R.id.tvRole);
        mTvFullname.setText(intent.getStringExtra("fullName"));
        mTvEmail.setText(intent.getStringExtra("email"));
        //mTvRole.setText(intent.getStringExtra("role"));


        mSignOutButton = (Button) findViewById(R.id.signOutButton);
        mSignOutButton.setOnClickListener(new View.OnClickListener(){
            @Override
            public void onClick(View v){
                FirebaseAuth.getInstance().signOut();
                Intent signIn = new Intent(UserDataActivity.this,SignInActivity.class);
                startActivity(signIn);
                finish();
            }
        });
    }
}

File: app/src/main/java/com/fpondarts/foodie/controller/UserDataActivity.kt
package com.fpondarts.foodie.controller

import androidx.appcompat.app.AppCompatActivity

import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.TextView

import com.fpondarts.foodie.R
import com.fpondarts.foodie.ui.auth.SignInActivity
import com.google.firebase.auth.FirebaseAuth

class UserDataActivity : AppCompatActivity() {

    private var mTvFullname: TextView? = null
    private var mTvEmail: TextView? = null
    //private TextView mTvRole;


    private var mSignOutButton: Button? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_user_data)

        val intent = intent
        mTvFullname = findViewById<View>(R.id.tvFullName) as TextView
        mTvEmail = findViewById<View>(R.id.tvEmail) as TextView
        //mTvRole = (TextView) findViewById(R.id.tvRole);
        mTvFullname!!.text = intent.getStringExtra("fullName")
        mTvEmail!!.text = intent.getStringExtra("email")
        //mTvRole.setText(intent.getStringExtra("role"));


        mSignOutButton = findViewById<View>(R.id.signOutButton) as Button
        mSignOutButton!!.setOnClickListener {
            FirebaseAuth.getInstance().signOut()
            val signIn = Intent(this@UserDataActivity, SignInActivity::class.java)
            startActivity(signIn)
            finish()
        }
    }
}
